### ISSUE TITLE
Admin date of birth doesn't factor in user locale set

### ISSUE THREAD

**Issue by extroniks** [magento/magento2#27027]
### Preconditions (*)

1. Magento 2.3.4 (but probably on previous versions also)
2. Login to admin
3. Have locale set to en_GB (UK) for the user

### Steps to reproduce (*)

1. Edit a customer
2. Select a date of birth after 12th of a month this year

### Expected result (*)

1. Date of birth is valid, you can save the customer

### Actual result (*)

1. Validation error happens "The Date of Birth should not be greater than today" because instead of DD/MM/YYYY the date is treated as MM/DD/YYYY

Here is a small composer patch I wrote for this:

```
--- a/view/base/web/js/lib/validation/rules.js
+++ b/view/base/web/js/lib/validation/rules.js
@@ -1069,12 +1069,12 @@ define([
             $.mage.__('This link is not allowed.')
         ],
         'validate-dob': [
-            function (value) {
+            function (value, param, params) {
                 if (value === '') {
                     return true;
                 }

-                return moment(value).isBefore(moment());
+                return moment.utc(value, params.dateFormat).isSameOrBefore(moment.utc());
             },
             $.mage.__('The Date of Birth should not be greater than today.')
         ]
--
2.17.1
```

**Comment by extroniks** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @extroniks. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @extroniks, here is your [Magento instance](https://i-27027-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-27027-2-4-develop.instances.magento-community.engineering/admin_200b](https://i-27027-2-4-develop.instances.magento-community.engineering/admin_200b)
Login: `38c1e601` Password: `0091173fa523`
Instance will be terminated in up to 3 hours.

**Comment by extroniks** [comment]
Reproduced on 2.4:
![image](https://user-images.githubusercontent.com/16757761/75337295-ffe3b400-588c-11ea-96c4-823e7247a756.png)

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @sudheers-kensium 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-31918` were created

**Issue Available**: @sudheers-kensium, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by webalvk** [comment]
Hi I'm having the same issue with local set to pt_br

**Comment by extroniks** [comment]
> Hi I'm having the same issue with local set to pt_br and magento 2.3.3, is there a fix yet?

You can use the patch I wrote, it should work fine, if you need the whole patch I can forward you one until an official fix is done.

**Comment by webalvk** [comment]
> 
> 
> @webalvk
> 
> > Hi I'm having the same issue with local set to pt_br and magento 2.3.3, is there a fix yet?
> 
> You can use the patch I wrote, it should work fine, if you need the whole patch I can forward you one until an official fix is done.

Thanks for fast replying... I'm not a programmer and i don't know how to apply this fix, could you provide me instructions? 

My email: web@alvk.com.br
Thanks

**Comment by nuzil** [comment]
@extroniks maybe you want to provide PR with the fix?

**Comment by deep701** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @deep701. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @deep701, here is your [Magento instance](https://i-27027-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-27027-2-4-develop.instances.magento-community.engineering/admin_7eeb](https://i-27027-2-4-develop.instances.magento-community.engineering/admin_7eeb)
Login: `ef55fb98` Password: `79769dc1fcf5`
Instance will be terminated in up to 3 hours.

**Comment by deep701** [comment]
@magento Unable to reproduce in latest instance.
![dateofbirth](https://user-images.githubusercontent.com/61613793/75604314-43445980-5afd-11ea-88a3-22845d7fe07b.png)

**Comment by keval123** [comment]
@magento  give me 2.3.4-develop instance

**Comment by extroniks** [comment]
@deep701  You need to select February after 12th. From your screenshot the date format is mm/dd/YYYY, issue happens when format is dd/mm/YYYY.

@nuzil 
Will see if I get the time during this week for a proper pull request / fix.

**Comment by extroniks** [comment]
@webalvk
Will send you the proper composer patch these days.

**Comment by vitmarks** [comment]
Hello same issue here, can you provide me also with the fix?

**Comment by NeilofSteel** [comment]
@extroniks Solution not working for me... wasn't there another fix few days ago? I used another fix which worked for me but I can't find it

**Comment by magento-engcom-team** [comment]
Hi @extroniks. Thank you for your report. 
The issue has been fixed in magento/magento2#27149 by @sergiy-v in 2.4-develop branch
Related commit(s):
 - [8f81132dcf27bc1b09ba89a6e7002a2f3dd02ef2](https://github.com/magento/magento2/commit/8f81132dcf27bc1b09ba89a6e7002a2f3dd02ef2)

The fix will be available with the upcoming 2.4.0 release.

**Comment by bcok22** [comment]
overwrite 

vendor/magento/module-customer/view/frontend/web/js/validation.js in your theme

and change

return moment(value, ["dd.MM.Y"]).isBefore(moment());

to

return moment(value, ["MM.dd.Y"]).isBefore(moment());
